src/Workspaces/Core/Portable/CodeFixes/SyntaxEditorBasedCodeFixProvider.cs

```diff
@@ -27,7 +27,7 @@ public sealed override FixAllProvider GetFixAllProvider()
             return FixAllAsync(document, ImmutableArray.Create(diagnostic), cancellationToken);
         }
 
-        private Task<Document> FixAllAsync(
+        protected virtual Task<Document> FixAllAsync(
```

Comment:
why is this being done... this feels very wonky to me.  the intent here was to hide this functionality and to force people through the syntax editor.  exposing this just makes this suffer from the problem inherent a lot in class hierarchies where excessive use of virtual extension points makes it super unclear:
1. what's going on.
2. what you're supposed to/allowed to override.

This is similar to the problem in 'CodeAction' where there are so many virtual inheritance points, and it's super difficult to know which ones are appropriate to overload and when

Reply:
I needed a hook to preprocess the document to add annotations and used the annotated document for the fix. I have reverted this and used a different approach.

---
In reply to: [230606341](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606341) [](ancestors = 230606341)